client/state/jetpack/modules/reducer.js

```diff
@@ -16,53 +15,53 @@ import {
 import { combineReducers } from 'calypso/state/utils';
 
 const createItemsReducer = ( active ) => {
-	return ( state, { siteId, moduleSlug } ) => {
-		return merge( {}, state, {
-			[ siteId ]: {
-				[ moduleSlug ]: {
-					active,
-				},
+	return ( state, { siteId, moduleSlug } ) => ( {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ moduleSlug ]: {
+				...state[ siteId ]?.[ moduleSlug ],
+				active,
 			},
-		} );
-	};
+		},
+	} );
 };
 
 const createItemsListReducer = () => {
-	return ( state, { siteId, modules } ) => {
-		return merge( {}, state, {
-			[ siteId ]: modules,
-		} );
-	};
+	return ( state, { siteId, modules } ) => ( {
+		...state,
+		[ siteId ]: modules,
+	} );
 };
```

Comment:
The `create*` functions here are apparently an artefact of a previous `createReducer` implementation. For example, there's no reason why this reducer should be created on every action dispatch. Constant `const itemsListReducer = ( state, action ) => ...` should be enough.

Many of the reducers here would benefit from `keyedReducer`. As always, it's completely up to you how many of these suggestions you want to implement, in this PR or in others.